packages/amplify_datastore/test/amplify_datastore_delete_test.dart

```diff
@@ -35,37 +37,37 @@ void main() {
   });
 
 
-  test('delete returns 1 sucessful result (no query predicate)', () async {
+  test('delete with a valid model executes without an error ', () async {
+    var json = await getJsonFromFile('delete_api/request/instance_no_predicate.json');
 
-    var json = await getJsonFromFile('delete_api/response/1_deleted_result.json');
+    dataStoreChannel.setMockMethodCallHandler((MethodCall methodCall) async {});
 
-    dataStoreChannel.setMockMethodCallHandler((MethodCall methodCall) async {
-      return json;
-    });
+    Post instance = Post(title: json["model"]["title"], rating: json["model"]["rating"], created: DateTime.parse(json["model"]["created"]), id: json["model"]["id"]);
+    bool executed;
 
-    Post instance = Post(title: json["serializedData"]["title"], rating: json["serializedData"]["rating"], created: DateTime.parse(json["serializedData"]["created"]), id: json["id"]);
+    try {
+      await dataStore.delete(instance);
+      executed = true;
+    } catch (e) {
+      executed = false;
+    }
 
-    Post post = await dataStore.delete(instance);
-    expect(post.id, instance.id);
-    expect(post.title, instance.title);
-    expect(post.rating, instance.rating);
-    expect(post.created, instance.created);
+    expect(executed, true);
```

Comment:
There is a `completes` predicate in the Dart `test_api` module. I think you can get rid of your `try`-`catch`, and this Boolean value. You might only need to do:
```dart
expect(() => dataStore.delete(instance), completes);
```
Refer: https://pub.dev/documentation/test_api/latest/test_api/completes.html

Reply:
Great suggestion - I got it working with this:
```
    Future<void> deleteFuture = dataStore.delete(instance);

     expect(deleteFuture, completes);
```